modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java

```diff
@@ -174,4 +178,21 @@
                       | WordDelimiterGraphFilter.STEM_ENGLISH_POSSESSIVE, null)));
         return filters;
     }
+
+    @Override
+    public List<PreConfiguredTokenizer> getPreConfiguredTokenizer() {
+        List<PreConfiguredTokenizer> tokenizers = new ArrayList<>();
+        tokenizers.add(PreConfiguredTokenizer.singleton("lowercase", LowerCaseTokenizer::new, () -> new TokenFilterFactory() {
+            @Override
+            public String name() {
+                return "lowercase";
```

Comment:
I wonder if (as a followup) `name()` could be removed from TokenFilterFactory? It seems to only have 3 uses, one of which is a test. Then TokenFilterFactory could be a functional interface and this would look much cleaner:

```
tokenizers.add(PreConfiguredTokenizer.singleton("lowercase", LowerCaseTokenizer::new, LowerCaseTokenFilter::new));
```